docs: Add documentation for array type(s) (#110)
tableau/hyper-db: website/docs/sql/datatype/index.md

# Data Types

Hyper provides a rich set of native data types.

> Generally, Hyper's type system can be divided into two buckets: _atomic_ types which describe single values, and _composite_ types which describe collection of values.

Reviewer: ```suggestion
Generally, Hyper's type system can be divided into two buckets: _atomic_ types, which describe single values, and _composite_ types, which describe collections of values.
```

Author: [9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)

> However, this distinction is made for educational puproses only; both kinds are equally supported, and there is no fundamental limitation applying to either category.

Reviewer: ```suggestion
However, this distinction is made for educational purposes only; both kinds of types are equally supported, and there is no fundamental limitation applying to either category.
```

Author: [9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)

## Atomic Types

> Atomic Types comprise fundamental, general-purpose data types.

Author: [c5c2486](https://github.com/tableau/hyper-db/pull/110/commits/c5c24866c3c874065963c3910fb97bddebdaa276)

> The following table lits all available atomic types.

Reviewer: ```suggestion
The following table lists all available atomic types.
```

Author: [9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)

> Most of the alternative names listed in the "Aliases" column are supported for compatibility with Postgres.

Reviewer: ```suggestion
Most of the alternative names listed in the "Aliases" column are supported for compatibility with PostgreSQL.
```

Author: [9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)

Up until Hyper API release [0.0.18825](/docs/releases#0.0.18825) Hyper used 64-bit floating points for all float types (i.e., also for `REAL`).
:::

## Composite Types

> Composite types are collections of data in a single SQL value.

Reviewer: ```suggestion
Composite types are collections of multiple data values in a single SQL value.
```

Author: [9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)

> They allow for dedicated [schema denormalization][schema-denormalization] which can be useful for specific domains, such as machine learning applications.

Reviewer: ```suggestion
They allow for dedicated [schema denormalization][schema-denormalization], which can be useful for specific domains, such as machine learning applications.
```

Author: [9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)
